### PR TITLE
Updated Group Member Add From Url block to enhance security and provide role defaults

### PR DESCRIPTION
**Pull request by cabal95**
# Contributor Agreement
_Have you filled out and sent your [Spark Contributor Agreement](http://www.rockrms.com/Content/RockExternal/Misc/Contributor%20Agreement.pdf) to secretary [at] sparkdevnetwork.org?_

**Yes**

# Context
_What is the problem you encountered that lead to you creating this pull request?_

Potential security concern when using this block. The Person is passed as a GUID but the Group is passed as a integer ID number. Meaning the user can modify the URL to easily add themselves to other Rock Groups that have common numbers out of the box - such as the Admin security role.

Also (I know, I'm a bad boy for not doing two merges) added an additional default member role if none is specified as the GroupType's defined default member role (if it is defined).

# Goal
_What will this pull request achieve and how will this fix the problem?_

Add optional layer of security to limit the group types allowed to be used.

# Strategy
_How have you implemented your solution?_

New Block Setting of `Limit Group Type` as a multi-select checkbox list that allows the Admin to optionally select which group types to limit the functionality to. For example if `Small Group` is selected then only groups of type Small Group will be allowed to be used.

# Possible Implications
_What could this change potentially impact? Are there any security considerations? Where could this potentially affect backwards compatibility?_

On the security side, no backwards compatibility issues. Default is to allow all group types.

On the member role, very minimal impact if the block was not configured correctly or the URL parameters were not passed correctly then they might get the group type's default member role instead of an error.

# Screenshots
_Provide us some screenshots if your pull request either alters existing UI or provides new UI. Arrows and labels are helpful._

n/a

# Documentation
_If your change effects the UI or needs to be documented in one of the existing [user guides](http://www.rockrms.com/Learn/Documentation), please provide the brief write-up here:_

n/a

### CODE REVIEW

**Comment by nairdo** [comment]
@cabal95 Seeing as how this change is similar to [this](https://github.com/SparkDevNetwork/Rock/commit/6e313a969f44fd9b3d0c3344955806d73da3b609) and [this](https://github.com/SparkDevNetwork/Rock/commit/12bb8b8bc2bd65c8a55b0bfcf3b5e2d5804b1695) change with the GroupRegistration block, have you considered the possibility of adding the other feature (Enable Passing Group Id) to this block as well?

**Comment by cabal95** [comment]
@nairdo Do you mean a way to disable providing a Group via the URL so it forces to only use the default group configured?  I can through that in. Shouldn't be too hard and could see that being a useful extension to self service kiosks.

**Comment by nairdo** [comment]
@cabal95 Well, perhaps that _and_ the ability to enforce that the group's Guid be passed instead of by Id like the other block does with the Enable Passing Group Id.  I'm just thinking security feature parity between these two blocks is ideal since they are so closely related in base functionality.

**Comment by cabal95** [comment]
Okay gotcha. I just glanced over the other two commits you mentioned I didn't dig in too deep. I'll see about adding another commit that brings parity closer to the other block.

**Comment by cabal95** [comment]
@nairdo Second commit provided for supporting `GroupGuid` in the Query String as welll as block setting to disallow `GroupId`.

**Comment by cabal95** [comment]
@nairdo Also, I'm still vague on the hotfix migrations, like the one in the commit you referenced and when those need to be included. So one may or may not need to be built for this change.

File: RockWeb/Blocks/Groups/GroupMemberAddFromUrl.ascx.cs [hunk 1]
```diff
@@ -118,6 +129,20 @@ protected override void OnLoad( EventArgs e )
                     return;
                 }
 
+                // Validate the group type.
+                if ( !string.IsNullOrEmpty( GetAttributeValue( "LimitGroupType" ) ) )
+                {
+                    bool groupTypeMatch = GetAttributeValue( "LimitGroupType" )
+                        .Split( ',' )
+                        .Contains( group.GroupType.Guid.ToString(), StringComparer.CurrentCultureIgnoreCase );
+
```
Comment (azturner) [inline RockWeb/Blocks/Groups/GroupMemberAddFromUrl.ascx.cs]: @cabal95 This request looks good. Thanks! In future you could use something like this here: 
`GetAttributeValue( "LimitGroupType" ).SplitDelimitedValues().AsGuidList().Contains( group.GroupType.Guid );`

Comment (cabal95) [inline RockWeb/Blocks/Groups/GroupMemberAddFromUrl.ascx.cs]: Can do. I'll try to remember those two methods!

**Pull request merged**